website/docs/d/ssm_parameter.html.markdown

```diff
@@ -29,7 +29,8 @@ The following arguments are supported:
 
 * `name` - (Required) The name of the parameter.
 * `with_decryption` - (Optional) Whether to return decrypted `SecureString` value. Defaults to `true`.
-
+* `with_default` - (Optional) Whether to return a default value if no parameter can be found. Defaults to `false`.
```

Comment:
I think `with_default` is not needed. Don't provide `default` if you don't need to get a default value back.

Reply:
Hi, I'm unsure how this can be implemented without breaking the actual behavior of failing if a SSM parameter cannot be found in SSM. Thats the reason why i created this "with_default" parameter.

If I skip "with_default" and set the Default value of "default" to be an empty string, I currently have no idea how to catch that in my code without returning a default value every time the parameter cannot be found. Any idea?

Reply:
Never mind, I figured it out by my self.